src/utils/image.js

```diff
@@ -363,41 +363,105 @@ export class RawImage {
+            width = Math.round((height / this.height) * this.width);
         } else if (nullish_height) {
-            height = (width / this.width) * this.height;
+            height = Math.round((width / this.width) * this.height);
         }
 
         if (IS_BROWSER_OR_WEBWORKER) {
-            // TODO use `resample` in browser environment
-
             // Store number of channels before resizing
             const numChannels = this.channels;
 
-            // Create canvas object for this image
-            const canvas = this.toCanvas();
+            // Create the output array for the resized image
+            const resizedData = new Uint8ClampedArray(width * height * numChannels);
+
+            // Scale factors for mapping new dimensions back to original dimensions
+            const xScale = this.width / width;
+            const yScale = this.height / height;
 
-            // Actually perform resizing using the canvas API
-            const ctx = createCanvasFunction(width, height).getContext('2d');
+            switch (resampleMethod) {
+                case 'bilinear':
+                    // Iterate over each pixel in the new image.
+                    for (let y = 0; y < height; y++) {
+                        for (let x = 0; x < width; x++) {
+                            // Map new coordinates to original coordinates.
+                            const srcX = x * xScale;
+                            const srcY = y * yScale;
+
+                            // Calculate the surrounding pixels.
+                            // Ensure that the pixels are within the bounds
+                            // of the image.
+                            const x0 = Math.floor(srcX);
+                            const x1 = Math.min(x0 + 1, this.width - 1);
+                            const y0 = Math.floor(srcY);
+                            const y1 = Math.min(y0 + 1, this.height - 1);
+
+                            // Calculate fractional parts for interpolation.
+                            const dx = srcX - x0;
+                            const dy = srcY - y0;
+
+                            for (let c = 0; c < numChannels; c++) {
+                                // Get the values of the new pixel area.
+                                // Always multiply by the width because we
+                                // storing the data in a 1D array.
+                                // To get the second row, we must add a full
+                                // width, then adding the x offset.
+                                const topLeft     = this.data[(((y0 * this.width) + x0) * numChannels) + c];
+                                const topRight    = this.data[(((y0 * this.width) + x1) * numChannels) + c];
+                                const bottomLeft  = this.data[(((y1 * this.width) + x0) * numChannels) + c];
+                                const bottomRight = this.data[(((y1 * this.width) + x1) * numChannels) + c];
```

Comment:
I'm not sure if my comment makes sense, but basically if we had an area like this:
```
array
1 2 3 4 5 6 7 8 9 10 11 12

image
1  2  3  4
5  6  7  8
9 10 11 12
```
To get the element on the third row (`3 - 1` since it is zero indexed) and second column (`2`), we must multiply by the width (`4`), then add the column.  
In this case, we would do:
```
((row - 1) * width) + column = index
((3 - 1) * 4) + 2 = 10
```